community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/expressions/ToIntFunction.scala

```diff
@@ -37,16 +37,12 @@ case class ToIntFunction(a: Expression) extends NullInNullOutExpression(a) {
       v.longValue()
     case v: String =>
       try {
-        v.toLong
+        val d = BigDecimal(v)
```

Comment:
Got it. You're totally right (assuming though, that we don't want `toInt("1.2")` to be an error).